### DISCUSSION TITLE
A solution to the fwupd problem

### DISCUSSION THREAD

**Discussion by Gaturinho** [fwupd/fwupd#5219] [Q&A]
I found a solution to the fwupd issue (reported at https://github.com/fwupd/fwupd/issues/3037) and already closed, but still not resolved. As I learned from the forums, the admins closed the discussion only because of the excess of complaints, which were disturbing the discussion and making it unproductive. However, the reported problem remains undefeated and unresolved, and continues to affect systems, even in fwupd version 1.7.5. But I found a solution for it, and it's simple. Perhaps there is interest in reopening or at least testing and/or making the solution available.
     
    I was able to solve the problem by installing a recommended package from fwupd. The excessive notifications are gone now, and it's only showing up during system boot, and only one notification, just once. It's been a little over 2 weeks since I started having "blows" of notifications with 4, 5 or even more, notifications in a row, and the outbreaks were repeated at least once an hour, all announcing a "network event", which I thought be just disconnection and reconnection events from the wifi network. At first I thought it was just my router's unstable connection to the network; but the absurd and frequent amount of notifications made me worried. Upon closer examination of the notifications, I noticed that they all came with complex code that I discovered was the addressing of my network driver on the system. Searching the internet, for pages that reported similar warnings in content ( systemd[1]: fwupd-refresh.service: Failed with result 'exit-code'. ), I came to this site here and several others, all with the same problem and all without a solution, with everyone lost and trying things that are kind of random and not always really recommendable. And instead of testing solutions like disabling or masking the service, uninstalling fwupd, or changing configuration files that do who knows what... I decided to try my approach. Even because it wasn't just anything with a problem, but the network driver (that's what it implied... although it wasn't really that), and it was clear that if I lost the network driver, it would be quite possible that I wouldn't managed to repair the system afterwards. So, the approaches cited all seemed to me something like hanging yourself in order not to be killed. So I tried my own approach, based on my vast experience in installing dependencies by fingernail. After unsuccessfully trying to reinstall fwupd, checking all fwupd dependencies and trying to reinstall them - all without success - I turned to my aforementioned vast experience, and the bizarre tricks it taught me: In my experience, sometimes installing some simple -dev package of a certain dependency fixes its bugs. But it's NOT that -dev packages have some magical power to fix buggy depes; but it just so happens that installing certain additional packages - usually -devs - causes the installation of the new package to re-install and reset the buggy package settings - logically if the problem is not in the package code, but in its settings, this might solve . So I searched Synaptic for some "fwupd-dev" or something, and there were several devs, but none seemed to be fwupd dev; but when searching, I found that there was a package recommendation (right click on the package and select from the menu the option "recommended packages"). This brings up a list, if there are recommended packages; these recommended lists usually indicate the installation of a lot of useless junk. But sometimes there's fool's gold there. In this case, there really was: "gir1.2-fwupd-2.0", a gobject package. That could be it. Installed, and got rid of the problem But working for me doesn't mean it will work for others, because my system is a Linux Mint 21 "Vanessa" all customized, with a lot of git packages and running on monstrous hardware ( ASUS, 5th/6th generation i5, 16Gb DDR5, system installed in UEFI mode on a system-only SSD, and with a dedicated graphics card), and on top of that running with Tuned in performance mode, and 3 additional fans. So, it's not uncommon for a solution that works for me, slow down other machines. I feel obliged to say this caveat. So, if you want to try it, test carefully and report if there is a loss of speed or an increase in CPU and memory consumption. Please, everyone who wants to test, contribute your experience.

**Comment by Gaturinho** [comment]
There was an important occurrence that confirms my thesis: The day before, there was an update of fwupd, which went up from 1.7.5 to 1.7.9~22.04.1, and with that, fwupd returned to issuing hallucinated notifications, and even worse than before: more than 30 in a row, at a frantic pace. That is, each time there is an update of fwup, it will be necessary to reinstall the package gir1.2-fwupd-2.0, so that the reinstallation of gir1.2-fwupd-2.0 corrects the buggy configuration file. But this new occurrence suggests that the problem may not be the fwup code itself (unless the new version's code has also been buggy, which is unlikely or at least rare in the Linux world < but common in Windows > ). Anyway, if the problem is repeated in the new version, the problem is probably not with fwup itself, but with some direct dependency on it. It's 23! And the new version has an even longer list of recommendations. I'll try an experiment: I'll also install gir1.2-fwupdplugin-1.0 (another gobject package that works with gir1.2-fwupd-2.0, but it's not on the list of recommendations or suggested ones) and then I'll reinstall fwupd to see if notifications return. This will allow you to see if it has something to do with the gobject, or if the problem is really just with fwupd itself or one of its dependencies.

**Comment by Gaturinho** [comment]
I didn't have to wait another half hour. The persistent notifications kept repeating themselves, so I reinstalled gir1.2-fwupd-2.0

**Comment by superm1** [comment]
This issue is fixed in Ubuntu in `1.7.9-1~22.04.1` for Ubuntu 22.04, and `1.7.9-1~20.04.1` for Ubuntu 20.04.  Please upgrade to those.

**Reply by Gaturinho** [reply]
But I'm using Mint 21 "Vanessa" which is just based on Ubuntu 20.4 Focal Fossa, and fwupd is properly updated, but the problem returns forever. It is very distressing, because whenever the internet speed decreases, the system understands the speed problem as being a driver problem and keeps trying to reinstall and reconfigure it over and over again, and with that the internet completely drops. There is no stable connection possible if the router driver is being reconfigured almost continuously for as long as the outbreak lasts. And that's why some veterans suggested masking fwupd. My suggestion just decreased the amount of times the event keeps repeating -- for fwupd 1.7.9; for fwupd 1.7.5 it came to a complete halt. So, we should ask ourselves if the problem is with fwupd itself ( well, the exaggeration of notifications certainly is ), or if the problem is in the system confusing any critical decrease in connection speed with a driver error ?

**Comment by Gaturinho** [comment]
I just looked at the connection after an event, and I found problems with IPV6. This is what the system keeps trying to reconfigure - it keeps trying to reconfigure the route - and it keeps affecting fwupd, which "brilliantly" concludes that the problem is the wlx30b5c210d1f4 driver, and then follows a routine/ritual of disconnecting, reinstalling and reconfiguring the driver. It's not just reconfiguring the route, it's reconfiguring EVERYTHING. I've found that I can terminate such an event by going into Network Manager, opening the settings (going to <wifi> connection, going to connection "details", clicking "apply" ) and manually applying what's there, without configuring anything . Just putting to apply the settings that are already in use. But I have to repeat this boring ritual every time I start the system. By the way, the router is not mine, and it is the connection to it that is giving problems. So would setting IPV6 to static solve it? But that wouldn't solve the errors the system makes and wouldn't be useful for other people with the same problem...

     Note that I am in a difficult situation: The router is not mine, and the condominium would require me to hire a technician; and the technician will surely tell me to install Windows.

**Comment by superm1** [comment]
>      I just looked at the connection after an event, and I found problems with IPV6. This is what the system keeps trying to reconfigure - it keeps trying to reconfigure the route - and it keeps affecting fwupd, which "brilliantly" concludes that the problem is the wlx30b5c210d1f4 driver, and then follows a routine/ritual of disconnecting, reinstalling and reconfiguring the driver. It's not just reconfiguring the route, it's reconfiguring EVERYTHING. I've found that I can terminate such an event by going into Network Manager, opening the settings (going to <wifi> connection, going to connection "details", clicking "apply" ) and manually applying what's there, without configuring anything . Just putting to apply the settings that are already in use. But I have to repeat this boring ritual every time I start the system. By the way, the router is not mine, and it is the connection to it that is giving problems. So would setting IPV6 to static solve it? But that wouldn't solve the errors the system makes and wouldn't be useful for other people with the same problem...
> 
>      Note that I am in a difficult situation: The router is not mine, and the condominium would require me to hire a technician; and the technician will surely tell me to install Windows.
> 

Your comments are VERY DIFFICULT to read. You are marking them as code which causes wrapping to be set incorrectly.

But I don't believe there is anything upstream fwupd can do about your conundrum. Perhaps you should just mask fwupd-refresh.service and run it manually from time to time.

**Reply by Gaturinho** [reply]
Since the previous post, I assumed that superm1 couldn't be wrong, and I did additional checks which showed me that ALL fwupd issues are related to IPV6 issues, and I had got one thing wrong: It's not "reinstalling" driver, but continually restarting it -- reconfigures and reinstalls over and over again for no understandable reason. So the problem is with IPV6, not with fwupd itself, so it's time to close this thread here and apologize for all the mess I made. I even suggest that if there is the potential for me to cause more confusion for other people looking for answers, if it wouldn't be better to delete the entire post, or if it's worth keeping as is. I don't have enough experience on GitHub to know which is better. so much so that as the old-timer noted, I'm posting it wrong, making it hard to read. But I don't know why, all my texts are getting like this, and I don't know where I'm going wrong. Thanks for listening.

   As for IPV6, the problem occurs because the damn thing keeps disconnecting and resetting the connection, and it got a little better after I changed its configuration from automatic to "local only". But I confess I don't know what this really entails, and in any case, it's not fwupd's problem.

**Reply by Gaturinho** [reply]
It looks like I fixed the IPV6 issue, and it's not the fwupd issue, although it might look like it, because fwupd kept issuing warnings about the wi-fi driver. And it's DEFINITELY a Mint 21 bug, because it's ABSURD. Of all the stupid defects I've seen in my life, this is the worst, the most lysergic: The wi-fi was in powersave mode, and the absurdity here is that I'm using a desktop, and with Tuned in troughput-performance mode , on top of that... But who knows why, the system configured itself as if the machine were an unplugged laptop -- and this despite the fact that I disabled power management in the settings. -- I had to go directly into the configuration file and rewrite it:

sudo xed /etc/NetworkManager/conf.d/default-wifi-powersave-on.conf

 and change the option powersave=3 to powersave=0

So I decided it's best to keep the post, in case another Mint user has the same problem with fwupd because of power management. Also because it's definitely a bug, although not from fwupd. There are several reports about Linux Mint 20 and 21, wich wi-fi dropping steadily over the last few weeks, and all report changing powersave-on.conf as a solution, archive which shouldn't even exist on a desktop installation. But it seems that the wi-fi instability gets in the way more because it appears in the fwupd notifications, which are not very clear and give a false impression of being a problem with the wi-fi driver or fwupd.

I want to give a special thanks to the veteran Superm1, who, by pointing out that the problem was impossible in the current version, encouraged me to research better so I didn't leave loose ends !
